src/rgw/rgw_file.cc

```diff
@@ -1145,7 +1145,7 @@ namespace rgw {
       emplace_attr(RGW_ATTR_SLO_UINDICATOR, std::move(slo_userindicator_bl));
     }
 
-    op_ret = processor->complete(s->obj_size, etag, &mtime, real_time(), attrs, delete_at,
+    op_ret = processor->complete(s->obj_size, etag, &mtime, real_time(), attrs, *delete_at,
```

Comment:
It looks we might be dereferencing an empty `boost::optional` here. If that's true, following check should help:
```
delete_at ? *delete_at : 0
```